usaspending_api/search/v2/views/spending_by_award.py

```diff
@@ -283,43 +288,82 @@ def populate_response(self, results: list, has_next: bool) -> dict:
+                )
+            )
+            sort_by_fields = self.get_sort_by_fields()
+            sort_by_fields.append("award_id")
+            database_fields = self.get_database_fields()
+            base_queryset = self.constants["filter_queryset_func"](self.filters)
+            queryset = self.annotate_queryset(base_queryset)
+            queryset = self.custom_queryset_order_by(queryset, sort_by_fields, self.pagination["sort_order"])
+            queryset = queryset.values(*list(database_fields))[record_num - 1 : record_num]
+
+            if len(queryset) != 1:
+                return {}
+            results = [
+                self.date_to_epoch_millis(queryset[0].get(self.get_sort_by_fields()[0])),
+                queryset[0].get("award_id"),
+            ]
+            search = (
                 AwardSearch()
                 .filter(filter_query)
                 .sort(*sorts)
-                .extra(search_after=[self.last_value, self.last_id])[0 : self.pagination["limit"]]
+                .extra(search_after=[*results])[: self.pagination["limit"] + 1]
             )
-            if self.last_value and self.last_id
-            else (
+            response = search.handle_execute()
+            return response
+        search = (
+            (
                 AwardSearch()
                 .filter(filter_query)
-                .sort(*sorts)[
-                    ((self.pagination["page"] - 1) * self.pagination["limit"]) : (
-                        ((self.pagination["page"] - 1) * self.pagination["limit"]) + self.pagination["limit"]
-                    )
-                ]
+                .sort(*sorts)
+                .extra(
+                    search_after=[self.date_to_epoch_millis(self.last_record_sort_value), self.last_record_unique_id]
+                )[: self.pagination["limit"] + 1]
             )
+            if self.last_record_sort_value is not None and self.last_record_unique_id is not None
+            else (AwardSearch().filter(filter_query).sort(*sorts)[record_num : record_num + self.pagination["limit"]])
         )
         response = search.handle_execute()
 
         return response
 
-    def construct_es_reponse(self, response) -> dict:
+    def construct_es_response(self, response) -> dict:
         results = []
         for res in response:
             hit = res.to_dict()
             row = {k: hit[v] for k, v in self.constants["internal_id_fields"].items()}
 
             for field in self.fields:
                 row[field] = hit.get(
-                    self.constants["elasticsearch_type_code_to_field_map"][hit[self.constants["award_semaphore"]]].get(
-                        field
-                    )
+                    str(
+                        self.constants["elasticsearch_type_code_to_field_map"][
+                            hit[self.constants["award_semaphore"]]
+                        ].get(field)
+                    ).replace(".keyword", "")
                 )
```

Comment:
ew. Sorry you had to do this ... dictionary-maths. Looks like a holdover from maybe a not-so good kind of polymorphism. 

(It doesn't help that I don't think `semaphore` is being used correctly here. Probably should just be `award_type_code` or `award_discriminator` or `award_partition_key`...)

But, since it's a non-obvious piece of code, please put a comment at the top. Like
`# Parsing API response values from ES query result JSON`

Reply:
`award_semaphore` is just a fancy name for ``award_type_flag`.
If the variable name accuracy is in question, it could be something like `award_type_code_fieldname`

Why could't the dict fields be modified cleanly beforehand like this?
https://github.com/fedspendingtransparency/usaspending-api/blob/dev/usaspending_api/awards/v2/lookups/elasticsearch_lookups.py#L91
```python
TRANSACTIONS_SOURCE_LOOKUP = {key: value.replace(".keyword", "") for key, value in TRANSACTIONS_LOOKUP.items()}
```

Reply:
Used Tony's suggestion of using a source lookup, removed the `.replace(".keyword","")`